### DISCUSSION TITLE
State estimation issue

### DISCUSSION THREAD

**Discussion by slazar394** [e2nIEE/pandapower#1829] [Q&A]
Hi everyone!

I'm currently working on a neural network for identifying measurements with large errors to improve the robustness of state estimation of power systems. Anyway, the neural network is working pretty well as I've achieved 95% accuracy on the test dataset, but I'm having trouble demonstrating the impact of bad measurement detection on the accuracy of state estimation. 

The first issue is that state estimation sometimes fails on normal measurements. I believe this has something to do with the way I set the standard deviation:

```python
    # Add the true measurements and perform state estimation
    measurements = base_measurements[:]
    for j in range(0, num_of_base_measurements):
        measurements[j].value = true_measurements[i, j]
        if measurements[j].meas_type == 'v':
            measurements[j].std = 0.01
        else:
            if measurements[j].value != 0:
                measurements[j].std = 0.01 * abs(measurements[j].value)
    net = add_measurements(net, measurements)
```
For reference, true measurements contain measurement noise generated as:

```python
        noise = np.random.normal(0, 0.01, net.measurement.value.shape)
        net.measurement.value += np.multiply(net.measurement.value, noise)
```

Second, my original idea was to simply set a large standard deviation for the measurements detected as erroneous, and I expected that this would improve the accuracy of the state estimate. By accuracy, I mean the RMSE between the correct and estimated states. However, this wasn't the case. To solve this problem, I simply removed the measurements that were identified as erroneous. Unfortunately, this sometimes leads to unobservability and the state estimate fails.

Does anyone have a clue as to what the problem might be with my approach to increasing the standard deviation? A code snippet is provided below.

```python
    # Identify the corrupted measurements, remove them from the measurements structure and perform state estimation
    predicted_label = torch.round(torch.sigmoid(model(X_test[i, ]))).detach().numpy()
    measurement_ids = np.where(predicted_label == 1)[0]
    # net.measurement.drop(labels=measurement_ids, axis=0, inplace=True)
    net.measurement.std_dev[measurement_ids] = 10000
```
Also, I've just recently switched to Python from MATLAB, so I hope I'm not making some stupid mistake 👍
